File: test_py/test_errors.py
from src_py.grndwork_api_client.errors import RequestError


def describe_request_error_to_string():
    def it_returns_error_name_and_message():
        error = RequestError('Bad Request')

        assert f'{error}' == 'RequestError: Bad Request'

    def it_returns_errors():
        error = RequestError('Bad Request', [
            {'message': 'Not a valid request'},
        ])

        assert f'{error}' == 'RequestError: Bad Request\nNot a valid request'

    def it_returns_errors_field():
        error = RequestError('Bad Request', [
            {'field': 'prop', 'message': 'Is required'},
        ])

        assert f'{error}' == 'RequestError: Bad Request\nField "prop" is required'

File: test_js/test_errors.ts
import {RequestError} from '../src_js/grndwork_api_client/errors';

describe('request error to string', () => {
  it('returns error name and message', () => {
    const error = new RequestError('Bad Request');

    expect(`${error}`).toEqual('RequestError: Bad Request');
  });

  it('returns errors', () => {
    const error = new RequestError('Bad Request', [
      {message: 'Not a valid request'},
    ]);

    expect(`${error}`).toEqual('RequestError: Bad Request\nNot a valid request');
  });

  it('returns errors field', () => {
    const error = new RequestError('Bad Request', [
      {field: 'prop', message: 'Is required'},
    ]);

    expect(`${error}`).toEqual('RequestError: Bad Request\nField "prop" is required');
  });
});
